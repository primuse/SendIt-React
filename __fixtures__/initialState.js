export const props = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: false
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: false
  }
};

export const props2 = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: true
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: true
  }
};

export const props3 = {
  parcel: {
    isLoading: true
  }
};
