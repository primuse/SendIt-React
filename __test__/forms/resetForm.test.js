import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import ConnectedResetForm, { ResetForm }
  from '../../src/components/forms/resetForm.jsx';
import { props } from '../../__fixtures__/initialState';

const onChange = jest.fn();
const resetPassword = jest.fn();
const mockStore = configureStore([thunk]);
const store = mockStore(props);
let wrapper;

describe('Reset Password form', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}>
        <ConnectedResetForm />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Reset password form', () => {
  beforeEach(() => {
    wrapper = mount(<ResetForm {...props}
      onChange={onChange} resetPassword={resetPassword} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should change state', () => {
    const input = wrapper.find('#email');
    input.instance().value = 'yinks@gmail.com';
    input.simulate('change');
    expect(wrapper.state().email).toEqual('yinks@gmail.com');
  });
  it('should simulate the onSubmit', () => {
    wrapper.find('form').simulate('submit');
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: 'cim@gmail.com' } });
    expect(resetPassword).toHaveBeenCalled();
  });
});
