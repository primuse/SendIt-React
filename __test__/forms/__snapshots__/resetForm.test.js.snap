// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password form renders correctly 1`] = `
<form
  className="text-center"
  id="resetForm"
  onSubmit={[Function]}
>
  <div
    className="mb-14"
  >
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
  </div>
  <div
    className="form-row"
  >
    <small>
      A reset link would be sent to you
    </small>
  </div>
  <div>
    <button
      className="btn sm is-outlined"
      type="submit"
    >
      Reset Password
    </button>
  </div>
</form>
`;
